addons/pos_self_order/models/res_config_settings.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api
+from werkzeug.urls import url_quote
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    pos_self_order_view_mode = fields.Boolean(
+        compute="_compute_pos_module_pos_self_order", store=True, readonly=False)
+    pos_self_order_image = fields.Image(
+        compute="_compute_pos_module_pos_self_order", store=True, readonly=False, max_width=1920, max_height=1080)
+    pos_self_order_image_name = fields.Char(
+        compute="_compute_pos_module_pos_self_order", store=True, readonly=False)
+
+    @api.depends("pos_config_id")
+    def _compute_pos_module_pos_self_order(self):
+        for res_config in self:
+            if not res_config.pos_config_id.self_order_view_mode:
+                res_config.update(
+                    {
+                        "pos_self_order_view_mode": False,
+                        "pos_self_order_image": False,
+                        "pos_self_order_image_name": False,
+                    }
+                )
+            else:
+                res_config.update(
+                    {
+                        "pos_self_order_view_mode": res_config.pos_config_id.self_order_view_mode,
+                        "pos_self_order_image": res_config.pos_config_id.self_order_image,
+                        "pos_self_order_image_name": res_config.pos_config_id.self_order_image_name,
+                    }
+                )
+
+    def generate_qr_codes_page(self):
+        """
+        Generate the data needed to print the QR codes page
+        """
+        no_of_qr_codes_per_page = 16
+        qr_codes_to_print = [
+            {
+                "id": 0,
+                "url": url_quote(f"{self.get_base_url()}/menu/{self.pos_config_id.id}"),
+            }
+            for i in range(0, no_of_qr_codes_per_page)
+        ]
+        data = {
+            "groups_of_tables": _splitListIntoNLists(qr_codes_to_print, 4),
+        }
+        return self.env.ref(
+            "pos_self_order.report_self_order_qr_codes_page"
+        ).report_action([], data=data)
+
+    def preview_self_order_app(self):
+        """
+        This function calls the preview_self_order_app function of the pos.config model
+        """
+        self.ensure_one()
+        return self.pos_config_id.preview_self_order_app()
+
+
+def _splitListIntoNLists(l, n):
+    """
+    Split a list into n lists
+    :param l: list to split
+    :type l: list
+    :param n: number of lists to split into
+    :type n: int
+    :return: list of lists
+    """
+    return [l[i: i + n] for i in range(0, len(l), n)]
```

Comment:
it's a one liner, you can directly use it in L50

Reply:
I could do it like this, but i don't think it would be very readable:
```
"groups_of_tables":   [qr_codes_to_print[i: i + n] for i in range(0, no_of_qr_codes_per_page, 4)] 
```